### PR TITLE
Feature 82 global interface

### LINKED ISSUE #82: Add global biotope object ts interface
Add an interface for the global biotope object describing `biotope.configuration` and similar.

### PR DESCRIPTION
**Pull request by dannystey**
rewrote the head.configuration.js and added it to biotope.ts 
why biotope.ts? because it can be extended in the future for other biotope related things and creates the global biotope namespace. (now with a biotope class)

closes #82

### CODE REVIEW

**Comment by timomayer** [comment]
like to have the namespace biotope, however in my oppinion we should use the configuration from the outside repo: https://github.com/biotope/biotope-configuration as a dependency?

**Comment by dannystey** [comment]
hm. i agree with you @timomayer. but this pr is an improvement of the current version anyway, isnt it? and the configuration can be moved to biotope-configuration as a next step (should create an issue for that). and the pr should fix the issue with the configuration interfaces ...

File: src/resources/ts/biotope.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	staticResourcesBase: string;
+	staticResourcesContentRepoBase: string;
+}
+
+interface biotopeConfigurationInterface {
+	biotope?: biotopeConfigurationBiotopeInterface;
+	data?: biotopeConfigurationDataInterface;
+}
+
+class biotopeConfiguration {
```
Comment (SheepFromHeaven) [inline src/resources/ts/biotope.ts]: Naming: Classes should always be upper camel case pls 🙏

File: src/resources/ts/biotope.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			dataValue[first] = this.setAux(rest.join('.'), value, dataValue[first]);
+		}
+		else {
+			dataValue[first] = value;
+		}
+		return dataValue;
+	}
+}
+
+class biotope {
```
Comment (SheepFromHeaven) [inline src/resources/ts/biotope.ts]: Naming: Classes should always be upper camel case pls 🙏

Comment (SheepFromHeaven) [reply src/resources/ts/biotope.ts]: In this case I also would rather have a name Like `BiotopeNamespace`

Comment (dannystey) [reply src/resources/ts/biotope.ts]: @SheepFromHeaven I am not sure? Why this postfix?

**Pull request merged**